crates/biome_yaml_parser/src/lexer/mod.rs

```diff
@@ -1,138 +1,235 @@
-#![allow(dead_code)]
+use std::collections::LinkedList;
 
 use biome_parser::{
     diagnostic::ParseDiagnostic,
-    lexer::{LexContext, Lexer, LexerCheckpoint, LexerWithCheckpoint, ReLexer, TokenFlags},
+    lexer::{Lexer, LexerCheckpoint},
 };
 use biome_rowan::{TextLen, TextRange, TextSize};
+use biome_unicode_table::{Dispatch::WHS, lookup_byte};
 use biome_yaml_syntax::{T, YamlSyntaxKind, YamlSyntaxKind::*};
-
-#[rustfmt::skip]
 mod tests;
 
 pub(crate) struct YamlLexer<'src> {
     /// Source text
     source: &'src str,
 
-    /// The start byte position in the source text of the next token.
-    position: usize,
-
-    /// If the source starts with a Unicode BOM, this is the number of bytes for that token.
-    unicode_bom_length: usize,
+    /// Where the lexer is in the source
+    current_coordinate: TextCoordinate,
```

Comment:
Are you keeping track of the document's indentation size anywhere?

Reply:
Yes we are, through this `current_coordinate`, which stores the current column, and [BlockScope](https://github.com/biomejs/biome/pull/6481/files#diff-0b7e31d5a26cd119870b271768a848f3590b4736f4207c1ad5d57dd183fea894R629). As stated above, since indentation (new line followed by spaces) alone is no longer enough to correctly determined whether the lexer is at the start of a new collection, it's now tracking the column of each tokens, and compares it with the left most column of the current scope.